### PR TITLE
feat(contract): update `SpaceOwner` token URI methods

### PR DESCRIPTION
**Pull request by shuhuiluo**
Added new methods to manage default URI in `SpaceOwner` and `SpaceOwnerUriBase`. This includes setting and getting the default URI, updating the `_render` method to use the default URI, and adding relevant events and validation checks.

Simplify layout access in `SpaceOwner` with direct getter and setter methods. Add selectors for `setDefaultUri` and `getDefaultUri` in deployment script. Extend tests to cover new `setDefaultUri` functionality and improve `tokenURI` tests.

Streamline the logic for building URIs by prioritizing space URIs and falling back to the default URI. The test suite has been updated to reflect these changes, ensuring that URIs with and without trailing slashes are handled correctly.

Added Solady library to the project and updated `remappings.txt` and `copyLibs.js` accordingly. Replaced OpenZeppelin's `Strings` with Solady's `LibString` for string operations in `SpaceOwnerUriBase.sol` and corresponding test file. Updated `package.json` and `yarn.lock` to include Solady.

https://linear.app/hnt-labs/issue/HNT-10522/space-metadata-uri
https://linear.app/hnt-labs/issue/HNT-10623/spec-town-metadata-changes-to-support-public-town-image

### CODE REVIEW

File: contracts/src/spaces/facets/owner/SpaceOwnerUriBase.sol [hunk 1]
```diff
@@ -7,37 +7,48 @@ import {ISpaceOwnerBase} from "contracts/src/spaces/facets/owner/ISpaceOwner.sol
...
-              Strings.toString(space.createdAt),
-              "}]}"
-            )
-          )
-        )
-      );
+    if (spaceAddress == address(0)) revert SpaceOwner__SpaceNotFound();
+
+    SpaceOwnerStorage.Space storage space = ds.spaceByAddress[spaceAddress];
+    string memory uri = bytes(space.uri).length == 0
```
[the remark this answers is not in the document]
Comment (shuhuiluo) [reply contracts/src/spaces/facets/owner/SpaceOwnerUriBase.sol:38]: @tak-hntlabs Could you confirm and update the Linear issue accordingly?

**Review by giuseppecrj — APPROVED** [review]

File: contracts/src/spaces/facets/owner/SpaceOwnerUriBase.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (tak-hntlabs) [reply contracts/src/spaces/facets/owner/SpaceOwnerUriBase.sol:38]: yes, Brian is right. If the caller sets it explicitly, we should respect that, and not add any thing else to the returned uri. I'll update the Linear.

File: contracts/src/spaces/facets/owner/SpaceOwnerUriBase.sol [hunk 2]
```diff
@@ -7,37 +7,48 @@ import {ISpaceOwnerBase} from "contracts/src/spaces/facets/owner/ISpaceOwner.sol
...
+      ? ds.defaultUri
+      : space.uri;
+
+    uint256 length = bytes(uri).length;
+    if (length == 0) revert SpaceOwner__DefaultUriNotSet();
+
+    unchecked {
+      // the ASCII code for "/" is 0x2f
+      if (bytes(uri)[length - 1] != 0x2f) {
+        return string.concat(uri, "/", Strings.toHexString(spaceAddress));
```
Comment (tak-hntlabs) [inline contracts/src/spaces/facets/owner/SpaceOwnerUriBase.sol:48]: nice 👍

**Review by tak-hntlabs — APPROVED** [review]

**Comment by giuseppecrj** [comment]
PROPOSAL: https://testnets.llama.xyz/orgs/river/base-sepolia/actions/159

**Pull request merged**